src/stream/src/executor/source/source_executor.rs

```diff
@@ -150,26 +153,17 @@ impl<S: StateStore> SourceExecutor<S> {
     }
 
     async fn refill_row_id_column(&mut self, chunk: StreamChunk, append_only: bool) -> StreamChunk {
-        let row_id_index = self.source_desc.row_id_index;
-
-        // if row_id_index is None, pk is not row_id, so no need to gen row_id and refill chunk
-        if let Some(row_id_index) = row_id_index {
-            let row_id_column_id = self.source_desc.columns[row_id_index as usize].column_id;
-            if let Some(idx) = self
-                .column_ids
-                .iter()
-                .position(|column_id| *column_id == row_id_column_id)
-            {
-                let (ops, mut columns, bitmap) = chunk.into_inner();
-                if append_only {
-                    columns[idx] = self.gen_row_id_column(columns[idx].array().len()).await;
-                } else {
-                    columns[idx] = self.gen_row_id_column_by_op(&columns[idx], &ops).await;
-                }
-                return StreamChunk::new(ops, columns, bitmap);
+        if let Some(idx) = self.row_id_index {
```

Comment:
The `row_id_index` is based on the `SourceDesc.columns`, while it's possible that we prune some columns and only keep those ones in `column_ids`. So seems this index cannot be directly used on columns of the chunk (which is already pruned in the connector).

Reply:
However, it's okay to remap this `index` when creating the source executor, instead of finding the position every time we call this function.